### ISSUE TITLE
Trash Compactor affects Death Star system

### ISSUE THREAD

**Issue by Gergall** [PlayersCommittee/gemp-swccg-public#736]
- [ ] Did you search through the list of **open issues** _first_ to ensure that there is not already an report for this issue?
Yes

## **Describe the bug**
A funny bug! When Trash Compactor is on table, LS is able to deploy to the Death Star SYSTEM without presence or Force icons.

## **To Reproduce**
1. Have Death Star system on table
2. Note that LS cannot deploy there yet
3. Deploy Trash Compactor
4. Suddenly, LS is able to deplyo to Death Star system!

## **Game Replay ID:**
https://gemp.starwarsccg.org/gemp-swccg/game.html?replayId=Gergall$qx80pqenji57i3pf
In this quick demonstration, the issue occurs on LS turn 2.

## **Expected behavior**
Trash Compactor should not affect the system.

## **Screenshots**
![image](https://github.com/PlayersCommittee/gemp-swccg-public/assets/6243118/352a0c1c-2662-48d3-973b-0bb5466a8474)

## **Additional context**
Trash Compactor site affects itself properly.  Problem is that it affects the system in addition to itself.
It doesn't seem to affect the other Death Star sites, so that's good.

**Comment by Gergall** [comment]
Additional note.  The bug only seems to allow an empty starship deploying with a pilot.

The card X-wing, which has a permanent pilot, is unable to deploy (which is correct).

Here is a screenshot showing that Falcon+Han is bugged, while X-wing is correct.
![image](https://github.com/PlayersCommittee/gemp-swccg-public/assets/6243118/cdb8dfac-b157-41ab-8ee7-087940cb4315)

**Comment by BOTVHD** [comment]
It would be really good to run the same sequence but have Captain Han in hand (can deploy to Falcon but not to Death Star) instead of Han Solo.  I'm curious if the issue that incorrectly allows simultaneous deployment is triggered by the pilot character being able to deploy to the related site.  Also worth checking if it also allows simultaneous deployment to an unrelated system (or site)

I should be able to set up some tests (in a mechanics/rules section for simultaneous deployment?) for these scenarios and identify the cause.

If we are lucky, the solution may even help solve some other situations where deployment is incorrectly allowed to cards related to the target, like #315

**Comment by EchoBaseTrooper** [comment]
> Somehow the game text for ignoring deployment restrictions on a site is getting extended to the related system - but only for simultaneous deployment of ship + pilot.

No, it is not because it is a related system. The ship+pilot simultaneous deployment only checks if any kind of deploying without presence/icons modifier applies to the pilot without checking any restrictions on that modifier. (see AbstractStarship.getValidDeployTargetWithPilotOrPassengerFilter)

**Comment by BOTVHD** [comment]
When fixing this issue, also should investigate #760 

This may be the same root cause (or at least closely related?).

**Comment by Gergall** [comment]
Testing needed

**Comment by billbisco** [comment]
Printed Death Star: Trash Compactor Light text is: you may deploy here without presence. If you control, Force drain +1 here.

"Here" is the site. The related Death Star system is a different location, so that text does not let you deploy to the system without presence or Force icons.

The hole was simultaneous empty starship + pilot. That path treated any "deploy without presence" modifier affecting the ship as a blanket ignore, so Trash Compactor unlocked the system for Falcon + a pilot. A permanent-pilot X-wing stayed blocked, which matches the report.

Fix: keep location-scoped deploy-without-presence per target instead of promoting it into a global ignore for simultaneous ship + pilot.

Tests lock Han to the site not the system, X-wing blocked at the system, and Falcon + Captain Han blocked at the system while still legal at a Cloud City docking bay.
